<?xml version="1.0"?>
<sdf version='1.6'>
<model name="mustard_bottle">
  <link name="base_link_mustard">

    <inertial>
      <pose frame=''>0 0 0 0 0 0</pose>
      <mass>0.603000</mass>
        <inertia>
          <ixx>0.001478</ixx>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyy>0.000633</iyy>
          <iyz>0</iyz>
          <izz>0.001770</izz>
        </inertia>
    </inertial>

    <visual name='base_link_mustard'>
      <pose frame=''>0 0 0 0 0 0</pose>
      <geometry>
        <mesh>
          <uri>../meshes/006_mustard_bottle_textured.obj</uri>
        </mesh>
      </geometry>
    </visual>

    <!--
      This places contact spheres on the corners of the visual box and a
      *slightly* smaller inset contact box (centered on the visual origin). This
      accounts for issues in the contact computation providing stable table
      contact *and* supports grasping.
      When the box is in stable contact with the ground plane, the corner
      spheres will provide fixed contact points (simulating distributed contact
      points around the face).  However, for arbitrary grip configuration, the
      slightly inset box will provide contact with a *slight* offset (in this
      case a deviation of 0.0005 m from the visual surface).
     -->
    <collision name='mustard_bottle_collision'>
      <pose>0 0.015 0 0 0 0</pose>
      <geometry>
        <box> 
          <size>0.090000 0.155300 0.052200</size>
        </box>
      </geometry>
      <material>
        <diffuse>0 1 0 1.0</diffuse>
      </material>
    </collision>

    <collision name='point_collision1'>
      <pose>0.045500 0.09315 0.026600 0 0 0 </pose>
      <geometry>
        <sphere><radius>1e-7</radius></sphere>
      </geometry>
    </collision>

    <collision name='point_collision2'>
      <pose>0.045500 -0.06315 0.026600 0 0 0 </pose>
      <geometry>
        <sphere><radius>1e-7</radius></sphere>
      </geometry>
    </collision>

    <collision name='point_collision3'>
      <pose>-0.045500 0.09315 0.026600 0 0 0 </pose>
      <geometry>
        <sphere><radius>1e-7</radius></sphere>
      </geometry>
    </collision>

    <collision name='point_collision4'>
      <pose>-0.045500 -0.06315 0.026600 0 0 0 </pose>
      <geometry>
        <sphere><radius>1e-7</radius></sphere>
      </geometry>
    </collision>

    <collision name='point_collision5'>
      <pose>0.045500 0.09315 -0.026600 0 0 0 </pose>
      <geometry>
        <sphere><radius>1e-7</radius></sphere>
      </geometry>
    </collision>

    <collision name='point_collision6'>
      <pose>0.045500 -0.06315 -0.026600 0 0 0 </pose>
      <geometry>
        <sphere><radius>1e-7</radius></sphere>
      </geometry>
    </collision>

    <collision name='point_collision7'>
      <pose>-0.045500 0.09315 -0.026600 0 0 0 </pose>
      <geometry>
        <sphere><radius>1e-7</radius></sphere>
      </geometry>
    </collision>

    <collision name='point_collision8'>
      <pose>-0.045500 -0.06315 -0.026600 0 0 0 </pose>
      <geometry>
        <sphere><radius>1e-7</radius></sphere>
      </geometry>
    </collision>

  </link>
</model>
</sdf>